packages/@lwc/features/src/flags.ts

```diff
@@ -4,25 +4,78 @@
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
-export type FeatureFlagValue = boolean | null;
-export type FeatureFlagLookup = { [name: string]: FeatureFlagValue };
+import { create, keys, defineProperty, isUndefined, isBoolean, globalThis } from '@lwc/shared';
+import { FeatureFlagMap, FeatureFlagName, FeatureFlagValue } from './types';
 
-const featureFlagLookup: FeatureFlagLookup = {
+const features: FeatureFlagMap = {
     ENABLE_REACTIVE_SETTER: null,
     ENABLE_HMR: null,
-    // Flag to toggle on/off the enforcement of innerText/outerText shadow dom semantic in elements when using synthetic shadow.
-    // Note: Once active, elements outside the lwc boundary are controlled by the ENABLE_ELEMENT_PATCH flag.
     ENABLE_INNER_OUTER_TEXT_PATCH: null,
-    // Flags to toggle on/off the enforcement of shadow dom semantic in element/node outside lwc boundary when using synthetic shadow.
     ENABLE_ELEMENT_PATCH: null,
     ENABLE_NODE_LIST_PATCH: null,
     ENABLE_HTML_COLLECTIONS_PATCH: null,
     ENABLE_NODE_PATCH: null,
-    // Disables the fix for #2121 where non-composed events are visible outside of their shadow root.
     ENABLE_NON_COMPOSED_EVENTS_LEAKAGE: null,
     ENABLE_LIGHT_DOM_COMPONENTS: null,
     ENABLE_MIXED_SHADOW_MODE: null,
 };
-export default featureFlagLookup;
 
-export { runtimeFlags, setFeatureFlag, setFeatureFlagForTest } from './runtime';
+if (!globalThis.lwcRuntimeFlags) {
+    Object.defineProperty(globalThis, 'lwcRuntimeFlags', { value: create(null) });
+}
+
+export const runtimeFlags: FeatureFlagMap = globalThis.lwcRuntimeFlags;
```

Comment:
This just got my attention: [FeatureFlagMap](https://github.com/salesforce/lwc/pull/2488/files#diff-edb353d559f765031e289098b1526914913512ed6d1ccfdff4f6b7465c65e851R16-R89) values are defined as `boolean | null` for all of the keys, but we are initializing it to `{}`, and [later on](https://github.com/salesforce/lwc/pull/2488/files#diff-55600cdf9b95a0c87924841b5567d768af07c00fdef7e1c05ce4907e3b8a7976R60) checking if the value is `undefined`. should we make it `Partial< FeatureFlagMap>`?

Reply:
It's a good call, I update the type to be `Partial<FeatureFlagMap>`. I think Typescript special case `undefined` check, this can explain why it isn't complaining.